### ISSUE TITLE
lerna run produces "stdout maxBuffer exceeded"

### ISSUE THREAD

**Issue by AlonMiz** [lerna/lerna#2690]
when running lerna run in parallel and stream i eventually get "stdout maxBuffer exceeded"
`lerna run ${command} --stream --parallel --max-buffer=999999999`
when debugging it looks like the maxBuffer injected to lerna options, but not to `get-stream` module

"lerna": "3.22.1"

## Expected Behavior
`max-buffer` should be used when using `--stream`

## Current Behavior
```
lerna sill argv {
lerna sill argv   _: [ 'run' ],
lerna sill argv   stream: true,
lerna sill argv   parallel: true,
lerna sill argv   'max-buffer': 999999999,
lerna sill argv   maxBuffer: 999999999,
lerna sill argv   loglevel: 'verbose',
lerna sill argv   scope: [ '@logz-prj/app-api', '@logz-prj/app-frontend' ],
lerna sill argv   lernaVersion: '3.22.1',
lerna sill argv   '$0': '/Users/***/***/gaia-hermes-ws/node_modules/.bin/lerna',
lerna sill argv   script: 'watch'
lerna sill argv }
lerna notice cli v3.22.1
lerna verb rootPath /Users/***/****/gaia-hermes-ws
lerna info versioning independent
lerna notice filter including ["@logz-prj/app-api","@logz-prj/app-frontend"]
lerna info filter [ '@logz-prj/app-api', '@logz-prj/app-frontend' ]
lerna info Executing command in 2 packages: "yarn run watch"
lerna sill npmRunScript.stream [ 'watch', [], '@logz-prj/app-api' ]
lerna sill getExecOpts /Users/****/****/gaia-hermes-ws/app-api undefined
```

when tracing the getStream function, the max buffer is the default 10M and not my specific 999999999
```
getStream { maxBuffer: 10000000 }
Trace
    at getStream (/Users/***/***/gaia-hermes-ws/node_modules/get-stream/index.js:21:10)
    at getStream (/Users/***/***/gaia-hermes-ws/node_modules/execa/index.js:135:9)
    at handlePromise (/Users/***/***/gaia-hermes-ws/node_modules/execa/index.js:270:3)
    at ChildProcess.spawned.catch (/Users/***/***/gaia-hermes-ws/node_modules/execa/index.js:313:32)
    at wrapError (/Users/***/***/gaia-hermes-ws/node_modules/@lerna/child-process/index.js:114:25)
    at Object.spawnStreaming (/Users/***/***/gaia-hermes-ws/node_modules/@lerna/child-process/index.js:66:10)
    at Function.stream (/Users/***/***/gaia-hermes-ws/node_modules/@lerna/npm-run-script/npm-run-script.js:27:32)
    at RunCommand.runScriptInPackageStreaming (/Users/***/***/gaia-hermes-ws/node_modules/@lerna/run/index.js:175:25)
    at /Users/***/***/gaia-hermes-ws/node_modules/@lerna/run/index.js:167:54
    at /Users/***/***/gaia-hermes-ws/node_modules/p-map/index.js:47:21
```

## Possible Solution
inject the maxBuffer value to `get-stream` module on run --stream

## Steps to Reproduce (for bugs)
1. lerna run watch --stream --parallel --max-buffer=999999999
2. have a big chunk of stdout
3. you'll eventually get "stdout maxBuffer exceeded"

<details><summary>lerna.json</summary><p>

```json
{
  "npmClient": "yarn",
  "useWorkspaces": true,
  "version": "independent"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
0 silly argv { _: [ 'run' ],
0 silly argv   stream: true,
0 silly argv   parallel: true,
0 silly argv   'max-buffer': 100000000,
0 silly argv   maxBuffer: 100000000,
0 silly argv   scope: '@logz-tools/kibana-server',
0 silly argv   lernaVersion: '3.22.1',
0 silly argv   '$0':
0 silly argv    '/Users/****/****/gaia-hermes-ws/node_modules/.bin/lerna',
0 silly argv   script: 'watch' }
1 notice cli v3.22.1
2 verbose rootPath /Users/****/****/gaia-hermes-ws
3 info versioning independent
4 notice filter including "@logz-tools/kibana-server"
5 info filter [ '@logz-tools/kibana-server' ]
6 info Executing command in 1 package: "yarn run watch"
7 silly npmRunScript.stream [ 'watch', [], '@logz-tools/kibana-server' ]
8 silly getExecOpts /Users/****/****/gaia-hermes-ws/tools/kibana-server undefined
9 error MaxBufferError: stdout maxBuffer exceeded
9 error     at PassThrough.stream.on (/Users/****/****/gaia-hermes-ws/node_modules/get-stream/index.js:41:19)
9 error     at PassThrough.emit (events.js:194:15)
9 error     at addChunk (_stream_readable.js:284:12)
9 error     at readableAddChunk (_stream_readable.js:261:13)
9 error     at PassThrough.Readable.push (_stream_readable.js:220:10)
9 error     at PassThrough.Transform.push (_stream_transform.js:151:32)
9 error     at PassThrough.afterTransform (_stream_transform.js:92:10)
9 error     at PassThrough._transform (_stream_passthrough.js:42:3)
9 error     at PassThrough.Transform._read (_stream_transform.js:190:10)
```
</p></details>

## Context
we suffer from this bug for some time, trying to get an alternative. 
basically slows us down when we develop

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.22.1 |
| `npm --version`  | 6.14.5 |
| `yarn --version` | 1.22.0 |
| `node --version` | v12.14.1 |

| OS | Version |
| --- | --- |
| macOS Catalina | 10.15.6 |

**Comment by njacob1001** [comment]
Same issue here

**Comment by Arrodl** [comment]
Any update on this? having the same error

**Comment by JamesHenry** [comment]
Hi Folks, I have taken a look into this, and I just want to understand if it is still an issues with latest version of lerna.

The latest version of lerna uses `execa` v5, which has a much larger 100MB max buffer (not the original 10MB mentioned in this and other issues).

Please could someone (e.g. @flotwig as you flagged this up) confirm they are breaching the 100MB cap? That seems like a very large amount of logs to be producing for a given invocation

**Comment by JamesHenry** [comment]
I was about to comment on the issue, but it makes sense to try and bring duplicate issues together. The other predates this one so I am closing this was in favour of it.

Please kindly let me know about usage with the latest version per my previous comment
